gcpim.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	imtypes "cloud-android-orchestration/api/instancemanager/v1"
+	compute "cloud.google.com/go/compute/apiv1"
+	"context"
+	"errors"
+	"github.com/google/uuid"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	// TODO(b/220891296): Make this configurable
+	projectId   = "google.com:cloud-android-jemoreira"
+	sourceImage = "projects/cloud-android-releases/global/images/cuttlefish-google-vsoc-0-9-21"
+	networkName = "projects/cloud-android-jemoreira/global/networks/default"
+)
+
+const (
+	namePrefix  = "cf-"
+	labelPrefix = "cf-"
+)
+
+var newUUIDString = func() string {
+	return uuid.New().String()
+}
+
+// GCP implementation of the instance manager.
+type GCPIM struct {
+	client *compute.InstancesClient
+}
+
+func NewGCPIM(client *compute.InstancesClient) *GCPIM {
+	result := &GCPIM{
+		client: client,
+	}
+	return result
+}
+
+func (m *GCPIM) DeviceFromId(name string, _ UserInfo) (DeviceDesc, error) {
+	return DeviceDesc{"127.0.0.1", "cvd-1"}, nil
+}
+
+func (m *GCPIM) InsertHost(zone string, req *imtypes.InsertHostRequest, user UserInfo) (*imtypes.Operation, error) {
+	if err := validateRequest(req); err != nil {
+		return nil, err
+	}
+	ctx := context.Background()
```

Comment:
From https://pkg.go.dev/context#TODO it seems context.TODO() is a better match for this.

Reply:
why TODO will be a better match here? documentation related to GCP client libraries always use context.Background() in their how-to documentation when making requests, an empty context could cause the requests to fail.

Reply:
Both Background and TODO say: " returns a non-nil, empty Context.".

TODO says: "Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter)."

Background says: "It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests."

It seems both do the same thing, but the one from Background is meant to be the first context ever created at the entry point (incoming requests, main function, etc) while TODO is meant to be used elsewhere when there is no surrounding context available, probably to make it easier to find once a surrounding context is available.

Reply:
good to know, thanks! done.